dspy/clients/lm.py

```diff
@@ -222,10 +225,26 @@ def litellm_text_completion(request, num_retries: int, cache={"no-cache": True,
         api_base=api_base,
         prompt=prompt,
         num_retries=num_retries,
+        **_get_litellm_retry_strategy_kwargs(),
         **kwargs,
     )
 
 
+def _get_litellm_retry_strategy_kwargs() -> Dict[str, str]:
```

Comment:
I fear this adds complexity that future maintainers of this file will not fully appreciate.

Maybe we just wait a bit and then just require recent litellm, i.e. >= 1.51.3?